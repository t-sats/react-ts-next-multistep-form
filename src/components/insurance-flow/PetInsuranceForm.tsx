import { useFormState } from "@/context/FormContext";
import AgeStep from "@/components/insurance-flow/AgeStep";
import EmailStep from "@/components/insurance-flow/EmailStep";
import SummaryStep from "@/components/insurance-flow/SummaryStep";

export const PetInsuranceSteps = () => {
    const { step } = useFormState();

    switch (step) {
        case 1:
            return <EmailStep/>;
        case 2:
            return <AgeStep/>;
        case 3:
            return <SummaryStep/>;
        default:
            return null;
    };
};

const PetInsuranceForm = () => {
    return <PetInsuranceSteps/>
};

export default PetInsuranceForm;
