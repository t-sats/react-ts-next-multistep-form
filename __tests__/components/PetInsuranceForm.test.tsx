import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import PetInsuranceForm from '@/components/insurance-flow/PetInsuranceForm';

test('shows an error for incorrect email value when the next button is clicked', () => {
  const { getByText, getByLabelText } = render(<PetInsuranceForm />);
  const emailInput = getByLabelText("Email:");
  const nextButton = getByText("Next");
  const error = "Please enter valid email";

  fireEvent.change(emailInput, { target: { value: "test" } });
  fireEvent.click(nextButton);

  expect(getByText(error)).toBeInTheDocument();
});

test('shows an error for empty email value when the next button is clicked', () => {
  const { getByText, getByLabelText } = render(<PetInsuranceForm />);
  const emailInput = getByLabelText("Email:");
  const nextButton = getByText("Next");
  const error = "Please enter valid email";

  fireEvent.change(emailInput, { target: { value: "" } });
  fireEvent.click(nextButton);

  expect(getByText(error)).toBeInTheDocument();
});

it('renders PetInsuranceForm component correctly', () => {
  const { container } = render(<PetInsuranceForm />)

  expect(container).toMatchSnapshot()
});
