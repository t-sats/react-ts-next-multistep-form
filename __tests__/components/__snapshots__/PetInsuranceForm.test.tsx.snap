// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PetInsuranceForm component correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-4 w-full"
  >
    <div
      class="flex flex-col gap-2 items-start"
    >
      <label
        for="email"
      >
        Email:
      </label>
      <input
        class="input input-primary w-full max-w-xs"
        id="email"
        name="email"
        type="email"
        value=""
      />
    </div>
    <button
      class="rounded-full border border-solid border-transparent bg-foreground text-background gap-2 hover:bg-[#28a745] font-medium text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 sm:w-auto"
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
